api/coreservice.go

```diff
@@ -1340,8 +1340,12 @@ func (core *coreService) LogsInBlockByHash(filter *logfilter.LogFilter, blockHas
 }
 
 // LogsInBlock filter logs in the block x
-func (core *coreService) LogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
-	return core.logsInBlock(filter, blockNumber)
+func (core *coreService) LogsInBlock(filter *logfilter.LogFilter, blockNumber uint64, c chan []*iotextypes.Log) {
```

Comment:
we may create a goroutine in `web3server` or `grpcserver`, instead of passing `c` for output purpose

Reply:
I can use a goroutine function literal to solve this issue.